.ci/new_ci.jenkinsfile

```diff
@@ -0,0 +1,435 @@
+        } else {
+            pr_number = detectPullRequestMerged()
+        }
+        assert pr_number != null, "Could not detect pull request number"
+
+        echo """
+            Build job information:
+            - branch_commit: ${branch_commit}
+            - merge_commit: ${merge_commit}
+            - pr_number: ${pr_number}
+            - pr_base_branch: ${pr_base_branch}
+            """
+    }
+
+    // Check for skip build
+    String scmTitle = sh(script: 'git log -1 --pretty=%B', returnStdout: true).trim()
+    if (scmTitle.contains('[skip ci]') || scmTitle.contains('[ci skip]')) {
+        currentBuild.result = 'SUCCESS'  // Keep success, so GitHub shows green status check
+        echo 'Aborting: Found [skip ci] on the commit message'
+        return  // Outside stage
+    }
+
+    List<Map> legacyImages = getLegacyImages(pr_number, "center-c3i-docker.jfrog.io")
+    echo "${legacyImages}"
+    List<Map> modernImages = getModernImages(pr_number, "center-c3i-docker.jfrog.io")
+    echo "${modernImages}"
+
+    if (!isMasterbranch) {
+        // Build Legacy Images
+        buildLegacyImages(legacyImages, "center-c3i-docker.jfrog.io", "center-c3i-docker")
+
+        // Build Modern Images
+        buildModernImages(modernImages, "center-c3i-docker.jfrog.io", "center-c3i-docker")
+    }
+
+    if (isMasterbranch) {
+        // Tag images as definitive (without pr_number)
+        legacyImages = tagImagesWithoutPRNumber(legacyImages, pr_number, "center-c3i-docker.jfrog.io", "center-c3i-docker")
+        modernImages = tagImagesWithoutPRNumber(modernImages, pr_number, "center-c3i-docker.jfrog.io", "center-c3i-docker")
+        // Promote to dockerhub
+        //promoteImages(legacyImages, "center-c3i-docker.jfrog.io", "center-c3i-docker", "docker.io", "docker-credentials")
+        //promoteImages(modernImages, "center-c3i-docker.jfrog.io", "center-c3i-docker", "docker.io", "docker-credentials")
+        // TODO: Clean images ffrom PR that are in Artifactory
+    }
+}
+
+def tagImagesWithoutPRNumber(List<Map> images, String pr_number, String registry, String credentialsID) {
+    withCredentials([usernamePassword(credentialsId: credentialsID, usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
+        sh 'docker login --username $USERNAME --password $PASSWORD ' + "${registry}"
+    }
+
+    images.each { Map image ->
+        // Pull image from desired registry
+        String registryImage = "${registry}/${image["name"]}"
+        sh "docker pull ${registryImage}"
+        // Remove pr_number from image tag
+        String imageWithoutPRNumber = image["name"].replace("-#${pr_number}", "")
+        echo "Tagging ${image["name"]} as ${imageWithoutPRNumber}"
+        // Tag image without pr_number and push it to desired registry
+        String registryImageWithoutPRNumber = "${registry}/${imageWithoutPRNumber}"
+        sh "docker tag ${registryImage} ${registryImageWithoutPRNumber}"
+        try {
```

Comment:
Please, separate this push, the method name is tag, but it actually does more than just a tag.

Reply:
you are right, but I want to tag the image in the server, not only locally. I would leave this for a future PR